tools/serverpod_cli/test/generator/dart/server_code_generator/inheritance_protocol_test.dart

```diff
@@ -0,0 +1,290 @@
+import 'package:analyzer/dart/analysis/utilities.dart';
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+
+import 'package:serverpod_cli/analyzer.dart';
+import 'package:serverpod_cli/src/analyzer/models/definitions.dart';
+import 'package:serverpod_cli/src/generator/dart/server_code_generator.dart';
+import 'package:serverpod_cli/src/test_util/builders/class_definition_builder.dart';
+import 'package:serverpod_cli/src/test_util/builders/endpoint_definition_builder.dart';
+import 'package:serverpod_cli/src/test_util/builders/generator_config_builder.dart';
+import 'package:serverpod_cli/src/test_util/compilation_unit_helpers.dart';
+
+const projectName = 'example_project';
+final config = GeneratorConfigBuilder().withName(projectName).build();
+const generator = DartServerCodeGenerator();
+
+void main() {
+  var expectedFileName = path.join(
+    'lib',
+    'src',
+    'generated',
+    'protocol.dart',
+  );
+
+  group('Given a sealed hierarchy when generating protocol files', () {
+    var parentClassName = 'Example';
```

Comment:
Question: How does this differentiate from the validation we are doing in the `simple` case in `sealed_class_test.dart`?

Reply:
This doesn't check the classes like in `sealed_class_test.dart` it tests the protocol file, to make sure we are not exporting sub-classes of a sealed hierarchy by themselves. 

And that sealed classes are not serialized